server/datastore/mysql/migrations/tables/20230719152138_ScheduledQueriesDataMigrator.go

```diff
@@ -0,0 +1,268 @@
+						p.pack_type   AS pack_type
+		FROM queries q
+				INNER JOIN scheduled_queries sq ON q.team_id IS NULL AND q.name = sq.query_name
+				INNER JOIN packs p ON sq.pack_id = p.id
+		WHERE p.pack_type <> 'global'
+		AND p.pack_type IS NOT NULL`
+
+	insertStmt = `
+		INSERT INTO queries (
+			team_id,
+			team_id_char,
+			name,
+			description,
+			query,
+			author_id,
+			saved,
+			platform,
+			min_osquery_version,
+			schedule_interval,
+			logging_type,
+			automations_enabled 
+		) VALUES ( ?, ?, ?, ?, ?, ?, ?, ?, ?, 1 )`
+
+	rows, err = tx.Query(selectStmt)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var query QueryWithScheduledFields
+		if err := rows.Scan(&query); err != nil {
+			return err
+		}
+
+		teamIDParts := strings.Split(query.PackType, "-")
+		teamID, err := strconv.Atoi(teamIDParts[1])
+		if err != nil {
+			return err
+		}
+
+		var loggingType string
+		if query.ScheduledQSnapshot != nil && *query.ScheduledQSnapshot {
+			loggingType = "snapshot"
+		}
+		if query.ScheduledQRemoved != nil {
+			if *query.ScheduledQRemoved {
+				loggingType = "differential"
+			} else {
+				loggingType = "differential_ignore_removals"
+			}
+		}
+		args := []interface{}{
+			teamID,
+			teamIDParts[1],
+			fmt.Sprintf("%s (Team Pack - %d)", query.Name, query.ScheduledQID),
+			query.Description,
+			query.Query,
+			query.AuthorID,
+			query.Saved,
+			query.ScheduledQPlatform,
+			query.ScheduledQVersion,
+			query.ScheduleInterval,
+			loggingType,
+		}
+		_, err = tx.Exec(insertStmt, args...)
+		if err != nil {
+			return err
+		}
+	}
+
+	// ----------------------------------------------------------------------------
+	// (2.2) Queries are non-scheduled and the author belongs to one or more teams:
+	// ----------------------------------------------------------------------------
+	type QueryWithTeamIDs struct {
+		fleet.Query
+		TeamRoles string `db:"team_roles"`
+	}
+
+	selectStmt = `
+		SELECT DISTINCT q.*, GROUP_CONCAT(CONCAT(ut.team_id, ':', ut.role)) AS team_roles
+		FROM queries q
+			LEFT JOIN scheduled_queries sq ON q.team_id IS NULL AND q.name = sq.query_name
+			INNER JOIN user_teams ut on q.author_id = ut.user_id
+		WHERE sq.id IS NULL
+		GROUP BY q.id`
+
+	insertStmt = `
+		INSERT INTO queries (
+			name,
+			description,
+			query,
+			author_id,
+			saved,
+			observer_can_run,
+			team_id,
+			team_id_char,
+		) VALUES ( ?, ?, ?, ?, ?, ?, ?, ? )`
+
+	rows, err = tx.Query(selectStmt)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var query QueryWithTeamIDs
+		if err := rows.Scan(&query); err != nil {
+			return err
+		}
+		teamRoles := strings.Split(query.TeamRoles, ",")
+		for _, teamRole := range teamRoles {
+			parts := strings.Split(teamRole, ":")
+			teamID, err := strconv.Atoi(parts[0])
+			if err != nil {
+				return err
+			}
+			role := parts[1]
+			if (role == "gitops") || (role == "observer" && !query.ObserverCanRun) {
```

Comment:
Let's check with product, I believe we want to move queries created by `gitops` too. @zhumo

Reply:
Yes, I think it doesn't matter where the query came from. We want all queries moved.

Reply:
Sounds good - thanks

Reply:
This is related to the following requirement:

> If a query was created by a team user (on one or more teams), then the query is duplicated as a team query on any teams the user is a maintainer of. The global query is kept (in case it's pointed at a 2017 "Pack").

We still want to do this, right? @zhumo 
My comment was around "team user that can create queries" can either be team admin, team maintainer or team gitops.